internal/plan/plan.go

```diff
@@ -90,7 +90,8 @@ type Service struct {
 	KillDelay      OptionalDuration         `yaml:"kill-delay,omitempty"`
 
 	// Log forwarding
-	LogTargets []string `yaml:"log-targets,omitempty"`
+	LogTargets        *LogTargets `yaml:"log-targets,omitempty"`
+	LogTargetsReplace *LogTargets `yaml:"^log-targets,omitempty"`
```

Comment:
It feels to me from the points above (if I am correct), that it would be better to shift the ```replace``` operand to the right-hand side of one a single ```LogTargets``` attribute, and make it only affect the log target list, not the keywords.

I also cannot help to observe that this could be a very common theme for attributes ... on service merge, we want to be able to allow attributes to opt in to replace.

Reply:
@flotter 1 is correct. 2, 3, 4 are open questions as to how it should behave here.

I think you're suggesting an approach similar to #221. We discussed the pros and cons of this. It is nicer having a single key, but the issue is that it makes it hard to programmatically write a Pebble layer e.g. inside a charm. You'd have to use a YAML library to write the `!replace` tag - you can't just pass in the Python list as you can in this approach.

Reply:
Sorry If I am repeating some decisions that was made - just ignore them :)